docs/articles/documentation/recipes/best-practices/best-practices.md

```diff
@@ -0,0 +1,223 @@
+        .click(interfaceSelect)
+        .click(interfaceOption.withText('JavaScript API'))
+        .expect(value).eql('JavaScript API')
+});
+```
+
+In this case, TestCafe applies the smart query mechanism and retries the assertion multiple times. This test responds to the changes that happen on the page and is more conclusive.
+
+## Use of Page Objects
+
+Use the Page Model in your tests. It helps you abstract the tested page into a single file that you can use to target page elements.
+
+This approach improves the flexibility of your tests - if the UI changes, change one file to update all the references. Define common operations as your model's methods. Tests are then more readable and less brittle because there is no duplicate code.
+
+The page model use is described in more detail in the [Page Model](../../guides/concepts/page-model.md) article.
+
+A code example is available in the [testcafe-examples](https://github.com/DevExpress/testcafe-examples/tree/master/examples/use-page-model) repository.
+
+## Use of Roles for Login
+
+Handle authentication during your tests with [User Roles](../../guides/advanced-guides/authentication.md#user-roles). See the [Authentication] article for more in-depth information.
+
+## File Structure
+
+Follow these guidelines to keep your test structure manageable and "clean":
+
+* If you use [page model](../../guides/concepts/page-model.md) for your tests (which is recommended), put all the page model files into one directory. If your application is divisible into logical chunks (components or subsystems), split up the associated page model objects into separate files.
+
+* Use the page model to store Selectors and compound actions that are common in your app. For instance, put all actions that are necessary to perform a login into one reusable function.
+
+> You can find an example of a page model in the [testcafe-examples](https://github.com/DevExpress/testcafe-examples/tree/master/examples/use-page-model) repository.
+
+* Use the `.testcaferc.json` configuration file to fine-tune your tests, which can be useful in CI/CD systems. This file must be located in the root directory of the project. For more information, read the [Configuration File](../../reference/configuration-file.md) article.
+
+* Define one `fixture` in every test file. While it's technically possible to define multiple, this is not recommended.
+
+* TestCafe tests are purely functional. As such, they should not be concerned about the implementation details and it is best to isolate them from production code. Keep your test files in a separate directory. You can name this directory appropriately (for instance, `tests`).
+
+* In your test folder, group your tests into subfolders. These groups cover different subsystems of your application with tests.
+
+* Any reused data (for example, large sets of expected values or forms inputs) is better stored in a directory of its own. Consider a descriptive folder name (for instance, `data`).
```

Comment:
I was arguing with the correctors about this, but now I tend to agree: `expected value` usually fits well in a very specific context only - when we are describing assertion API methods.

In a broader context, like here, saying *large sets of expected values* confuses.

You can think of this as our professional deformation. We strongly associate 'expected values' with assertions, while a regular native speaker can come up with tons of possible meanings of 'expected values' when not in a specific context.

Reply:
> Any reused data (for example, large sets of reference values or forms inputs) is better stored in a directory of its own. Consider a descriptive folder name (for instance, `data`).